scripts/gis-material-update/validate-deploy/modules/common.py

```diff
@@ -0,0 +1,91 @@
+import math
+from sqlalchemy import create_engine, text
+    
+def deploy(pg_conn_uri, tormays_table_org, tormays_table_temp, logger=None):
+    # validate data amount: is it between given limits
+    delete_result = None
+    engine = create_engine(pg_conn_uri)
+    with engine.connect() as connection:
+        transaction = connection.begin()
+        delete_sql = "delete from " + tormays_table_org
+        try:
+            logger.info("Deploy ...")
+            delete_result = connection.execute(text(delete_sql))
+            logger.info(tormays_table_org + ": " + f"{delete_result.rowcount} rows deleted.")
+        except Exception as e:
+            # Transaction implicitly rolls back if an exception occurred within the "with" block
```

Comment:
The exception needs to be raised when the `with` block ends for the rollback to happen. If the exception is handled in `except`-block, it's no longer raised and the transaction will be committed.

Reply:
After playing around with the transactions for a while I think I understand now: this works correctly, but not quite in the way these comments suggest.

1. There's no autocommit here, so the `transaction.commit()` needs to be called for the commit to happen. This wouldn't be the case if the transaction was defined as a `with` block.
2. If the connection ends without a commit, the transaction is rolled back.
3. If there's an error in the SQL, the database automatically rolls back the transaction.

So this works because we don't let the execution reach the commit unless both statements are executed without errors.